oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/profiling/trace/ProfileTaskMutationService.java

```diff
@@ -140,14 +140,16 @@ private String checkDataSuccess(final String serviceId,
 
         // Each service can monitor up to 1 endpoints during the execution of tasks
         long startTimeBucket = TimeBucket.getMinuteTimeBucket(monitorStartTime);
-        long endTimeBucket = TimeBucket.getMinuteTimeBucket(monitorEndTime);
         final List<ProfileTask> alreadyHaveTaskList = getProfileTaskDAO().getTaskList(
-            serviceId, null, startTimeBucket, endTimeBucket, 1);
+            serviceId, null, startTimeBucket, null, null);
```

Comment:
1. If you use the start time to filter, then you may not get any data. For example, you create a 10 min task at 10:00, then when you create a new task at 10:05, then you cannot get the existing task. Am I right?
2. You should limit the profiling task count.

Reply:
sorry, I used to think of TIME_BUCKE as END_TIME.
What do you think the "limit" should be set to.

Reply:
sorry, I used to think of TIME_BUCKE as END_TIME.
What do you think the "limit" should be set to.

Reply:
I think `1` is enough. Usually, you just need to check if the most recent task has overlap or not.